compiler/plugins/target/ROCM/Dialect/ROCM/IR/ROCMAttrs.cpp

```diff
@@ -79,6 +91,362 @@ static LogicalResult handleArgmaxUkernel(
+    op.emitWarning("Failed to serialize bitcode.");
+    return {};
+  }
+  llvm::LLVMContext llvmContext;
+  llvm::Expected<std::unique_ptr<llvm::Module>> module =
+      llvm::getLazyBitcodeModule(llvm::MemoryBufferRef{buffer, ukernelName},
+                                 llvmContext,
+                                 /*ShouldLazyLoadMetadata=*/true);
+  if (!module) {
+    op.emitWarning("Failed to parse bitcode module.");
+    return {};
+  }
+  llvm::EngineBuilder builder(std::move(module.get()));
+  std::string builderError;
+  builder.setEngineKind(llvm::EngineKind::Interpreter)
+      .setErrorStr(&builderError);
+  std::unique_ptr<llvm::ExecutionEngine> interpreter{builder.create()};
+  if (!interpreter) {
+    op.emitWarning("Failed to create the interpreter.");
+    return {};
+  }
+  std::string queryFuncName =
+      llvm::formatv("{}_query_shared_memory_bytes", ukernelName);
+  llvm::Function *func = interpreter->FindFunctionNamed(queryFuncName);
+  if (!func) {
+    op.emitWarning(llvm::formatv(
+        "Bitcode does not contain a function named {}.", queryFuncName));
+    return {};
+  }
+  auto constI32 = [](int32_t val) {
+    llvm::GenericValue v;
+    v.IntVal = APInt(32, val);
+    return v;
+  };
+  SmallVector<llvm::GenericValue> args{
+      constI32(mma.getIntrinsicsM()), constI32(mma.getSubgroupsM()),
+      constI32(mma.getIntrinsicsN()), constI32(mma.getSubgroupsN()),
+      constI32(mma.getIntrinsicsK())};
+  if (func->arg_size() != args.size()) {
+    op.emitWarning(
+        llvm::formatv("Bitcode function {} takes {} arguments. Expected {}.",
+                      queryFuncName, func->arg_size(), args.size()));
+    return {};
+  }
+  llvm::GenericValue interpreterResult = interpreter->runFunction(func, args);
+  if (interpreter->hasError()) {
+    op.emitWarning(llvm::formatv("Error while interpreting bitcode: {}.",
+                                 interpreter->getErrorMessage()));
+    return {};
+  }
+  int sharedMemoryBytes = interpreterResult.IntVal.getSExtValue();
+
+  // Reject a ukernel that would consume too much shared memory, which we need
+  // to save for other purposes. This threshold can always be adjusted but we
+  // default to a low threshold to get an early signal.
+  int maxSharedMemoryBytes = getSharedMemoryBytes(gpuTarget) / 4;
+  if (sharedMemoryBytes > maxSharedMemoryBytes) {
+    op.emitWarning(llvm::formatv("The shared memory size {} required by the "
+                                 "ukernel exceeds the maximum allowed size {}.",
+                                 sharedMemoryBytes, maxSharedMemoryBytes));
+    return {};
+  }
+  return sharedMemoryBytes;
+}
+
+// Returns the shared memory size required by the multi_mma ukernel.
+// On failure, an op warning is emitted and {} is returned.
+// Uses a static cache to avoid calling expensivelyEvaluateSharedMemoryBytes
+// more than once per DataTiledMMAAttr value.
+static std::optional<int>
```

Comment:
I know you just copied this, but maybe make sharedMemoryBytes consistently int64_t in this file. Here, it's an int, above it's int64_t..